WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -242,6 +255,8 @@ public ICoinsView AsAllCoinsView()
 
 	public ICoinsView Available() => AsCoinsView().Available();
 
+	public ICoinsView AvailableForCoinJoin() => new CoinsView(AsCoinsView().Where(coin => !CoinPrison.IsCoinBanned(coin, DateTimeOffset.Now, out _)));
```

Comment:
First, please make methods deterministic. You don't want to know if coins are available now, what you want is to now if coins will be available before input registration finishes so, you have to receive the datetime.
Second, never use `Now` in business code, always use `UtcNow`. `